File: FrontEnd/src/app/entities/filter.ts
import { IpRange } from "./ipRange";
import { LengthRange } from "./lengthRange";
import { Protocol } from "./protocol";

export class Filter {
  destinationIpRanges: IpRange[];
  sourceIpRanges: IpRange[];
  destinationIps: string[];
  sourceIps: string[];
  protocols: Protocol[];
  lengthRanges: LengthRange[];

  constructor() {
    this.destinationIpRanges = [];
    this.sourceIpRanges = [];
    this.destinationIps = [];
    this.sourceIps = [];
    this.protocols = [];
    this.lengthRanges = [];
  }
  
}

File: BackEnd/app/models/Filter.py
from pydantic import BaseModel
from typing import List
from models.IpRange import IpRange
from models.Protocol import Protocol
from models.LengthRange import LengthRange

class Filter(BaseModel):
    destinationIpRanges: List[IpRange] = []
    sourceIpRanges: List[IpRange] = []
    destinationIps: List[str] = []
    sourceIps: List[str] = []
    protocols: List[Protocol] = []
    lengthRanges: List[LengthRange] = []
